echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/orca/PipelineInitiator.java

```diff
@@ -136,62 +151,101 @@ public void startPipeline(Pipeline pipeline) {
+      } else {
+        // If we should not propagate authentication, create an empty User object for the request
+        User korkUser = new User();
+
+        if (fiatStatus.isEnabled()) {
+          if (pipeline.getTrigger() != null && pipeline.getTrigger().getRunAsUser() != null) {
+            korkUser.setEmail(pipeline.getTrigger().getRunAsUser());
+          } else {
+            // consistent with the existing pattern of
+            // `AuthenticatedRequest.getSpinnakerUser().orElse("anonymous")`
+            // and defaulting to `anonymous` throughout all Spinnaker services
+            korkUser.setEmail("anonymous");
+          }
+          korkUser.setAllowedAccounts(getAllowedAccountsForUser(korkUser.getEmail()));
+        }
+
+        response =
+            AuthenticatedRequest.propagate(() -> triggerWithRetries(pipeline), korkUser).call();
+      }
+
+      log.info("Successfully triggered {}: execution id: {}", pipeline, response.getRef());
+
+    } catch (RetrofitError e) {
+      String orcaResponse = "N/A";
+      int status = 0;
+
+      if (e.getResponse() != null) {
+        status = e.getResponse().getStatus();
+
+        if (e.getResponse().getBody() != null) {
+          orcaResponse = new String(((TypedByteArray) e.getResponse().getBody()).getBytes());
         }
-        korkUser.setAllowedAccounts(getAllowedAccountsForUser(korkUser.getEmail()));
       }
-      AuthenticatedRequest.propagate(orcaResponse::subscribe, korkUser).call();
+
+      log.error(
+          "Failed to trigger {} HTTP: {}\norca error: {}\npayload: {}",
+          pipeline,
+          status,
+          orcaResponse,
+          pipelineAsString(pipeline));
+
+      logOrcaErrorMetric(e.getClass().getName(), triggerSource.name());
+    } catch (Exception e) {
+      log.error(
+          "Failed to trigger {}\nerror: {}\npayload: {}", pipeline, e, pipelineAsString(pipeline));
+
+      logOrcaErrorMetric(e.getClass().getName(), triggerSource.name());
     }
   }
 
-  private Observable<OrcaService.TriggerResponse> createTriggerObservable(Pipeline pipeline) {
-    return orca.trigger(pipeline);
-  }
+  private TriggerResponse triggerWithRetries(Pipeline pipeline) {
+    int attempts = 0;
 
-  private void onOrcaResponse(TriggerResponse response) {
-    log.info("Triggered pipeline {}", response.getRef());
-  }
+    while (true) {
+      try {
+        attempts++;
+        return orca.trigger(pipeline);
+      } catch (RetrofitError e) {
+        if ((attempts >= retryCount) || !isRetryableError(e)) {
+          throw e;
+        }
+      }
 
-  private void onOrcaError(Pipeline pipeline, Throwable error) {
-    registry.counter("orca.errors", "exception", error.getClass().getName()).increment();
-    log.error("Error triggering pipeline: {}", pipeline, error);
-  }
+      try {
+        Thread.sleep(retryDelayMillis);
+      } catch (InterruptedException ignored) {
+      }
 
+      log.warn("Retrying pipeline trigger for {}", pipeline);
```

Comment:
log.warn("Retrying pipeline trigger (attempt {}/{}) for {}", attempts, retryCount, pipeline);